### PR TITLE
Remove field labels and change required text

### LINKED ISSUE #410: Don't display redundant form field labels
- Don't display redundant field labels 
- Retain field labels for screenreaders

### PR DESCRIPTION
**Pull request by neilmb**
# Remove redundant field labels #

## 🗣 Description ##

In our restructuring of the Django form templates we showed all of the form labels which is not appropriate for every page. This hides the field labels on some pages using the USWDS screenreader-only class tag (including some code to do so just for legends). Hiding field labels also made it necessary to put the indication that a question is "required" in a different place. In consultation with @igorkorenfeld this uses the sentence "**This question is required.**" where appropriate. It also hides the "Required fields are marked with an asterisk" text on pages where the asterisk is not used.

## 💭 Motivation and context ##

Closes #410

### CODE REVIEW

File: src/registrar/templates/application_org_election.html [hunk 1]
```diff
@@ -5,11 +5,16 @@
   <h2 class="margin-bottom-05">
     Is your organization an election office?
   </h2>
+
+  <p class="text-semibold">This question is required.</p>
 {% endblock %}
 
+{% block form_required_fields_help_text %}
+{# commented out so it does not appear on this page #}
+{% endblock %}
```
Comment (SSPJ) [inline src/registrar/templates/application_org_election.html:14]: My only overall suggestion would be to consider explicitly placing the asterisk on the visible "label":

```
Is your organization an election office?
  <abbr class="usa-hint usa-hint--required" title="required">*</abbr>
```

and displaying a consistent help text throughout the form (at least on the pages which have required questions). I think self-similar wording combined with a splash of red color will reduce cognitive load and help users with cognitive impairments. There's some [supplemental guidance](https://www.w3.org/WAI/WCAG2/supplemental/patterns/o1p03-consistent-design/) from WCAG, although I might be applying it in an overly broad manner.

I'll defer to @igorkorenfeld on this. Igor, could you consider the matter and leave a reply for Neil?

**Review by SSPJ — APPROVED** [review]
Nice work, Neil! LGTM 👍

Sorry that I didn't get this reviewed on Friday. 💦 

The Basics
- [x] I read the code
- [x] I ran the code
 
Documentation
- [x] README (or other documentation) is correct and updated
- [x] all new functions and methods are commented
    - [x] using plain language
- [x] licenses for new dependencies are recorded, if any
- [x] licenses for updated dependencies have been checked for accuracy, if any
- [x] new components have been added to the system diagram
 
Security
- [x] security false positives are documented
- [x] data from external sources is cleaned and clearly marked
- [x] significant user actions are logged to a database table
 
Reliability
- [x] good level of error handling
    - [x] interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests

Accessibility
- [x] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [x] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [x] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [x] All form inputs have explicit labels
    - [x] All relevant images use an img tag
    - [x] All images have alt attributes
    - [x] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [x] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [x] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [x] Tab order is logical
    - [x] Form instructions are associated with inputs
    - [x] Site doesn’t timeout unexpectedly. Identify elements that may “timeout” and verify that the user can request more time
    - [x] Tables are coded properly. Tables have proper headers and column attributes
    - [x] Headings are nested properly. Heading elements are nested in a logical way
    - [x] Frames are named. All frames have a name element
    - [x] Flashing elements are compliant. Elements that flash on screen do so at a rate of less than 3 Hz
    - [x] Language is set. The language for the page is set
    - [x] The language for sections on the page that differ from the site language are set
    - [x] CSS is not required to use the page. The page makes sense with or without CSS
    - [x] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [x] Page titles are descriptive
    - [x] Required plugins are linked on the page
 
Device Matrix
- [x] firefox
    - [x] web pages render correctly
    - [x] interactive functionality is operative
- [x] chrome/chromium/edge
    - [x] web pages render correctly
    - [x] interactive functionality is operative
- [x] web page is readable and usable 
    - [x] at 480px
    - [x] at 1024px

File: src/registrar/templates/application_org_election.html (hunk 1 above)
Comment (igorkorenfeld) [reply src/registrar/templates/application_org_election.html:14]: Thanks for tagging me in on this Seamus. 

Adding the asterisk to the question sounds good to me, if keep the "This question is required" text below it, closer to the input field (I wanted the proximity of the note to the input field). The only trouble is that I wouldn't want to have a legend for the asterisk in addition to the "This question is required."

**Comment by neilmb** [comment]
Reviewed the asterisk situation with @igorkorenfeld and @michelle-rago, merging as it is now. We'll see if users have anything to say about it in the next feedback rounds.

**Pull request merged**